### PR TITLE
dotnet log-trace-injection updates

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?
Updates docs "Connect Logs and Traces" for .NET 

### Motivation
Googledoc from PM

### Preview
https://docs-staging.datadoghq.com/kari/dotnet-logs-traces-update/tracing/connect_logs_and_traces/dotnet

Added Further Reading link to updated page:

- https://docs-staging.datadoghq.com/kari/dotnet-logs-traces-update/tracing/setup_overview/setup/dotnet-core?tab=windows#further-reading
- https://docs-staging.datadoghq.com/kari/dotnet-logs-traces-update/tracing/setup_overview/setup/dotnet-framework?tab=environmentvariables#further-reading

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by jaycdave88 — COMMENTED** [review]
This is great - thank you! I think the only part that we need would be to sub-link this back to the main .NET Framework and .NET Core footer section under Future Readings.

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 1]
```diff
@@ -75,61 +94,52 @@ For NLog version 4.6+:
...
+1. Reference the [`Datadog.Trace` NuGet package][10] in your project.
+
+2. Use the `CorrelationIdentifier` API to retrieve correlation identifiers and add them to the log context while a span is active.
+
+3. Configure mapped diagnostic context (MDC) to use the following injected keys:
+
+    | Required key   | Description                                  |
+    | -------------- | -------------------------------------------- |
+    | `dd.env`       | Globally configures the `env` for the tracer. Defaults to `""` if not set. |
+    | `dd.service`   | Globally configures the root service name. Ddefaults to the name of the application or IIS site name if not set.  |
```
Comment (lucaspimentel) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:132]:
```suggestion
    | `dd.service`   | Globally configures the root service name. Defaults to the name of the application or IIS site name if not set.  |
```

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 2]
```diff
@@ -75,61 +94,52 @@ For NLog version 4.6+:
...
+    | `dd.version`   | Globally configures `version` for the service. Defaults to `""` if not set.  |
+    | `dd.trace_id`  | Active trace ID during the log statement. Defaults to `0` if no trace.  |
+    | `dd.span_id`   | Active span ID during the log statement. Default to 0 if no trace. |
```
Comment (lucaspimentel) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:135]:
```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Default to `0` if no trace. |
```

Comment (kayayarai) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:135]:
```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Defaults to `0` if no trace. |
```

Comment (kayayarai) [inline content/en/tracing/connect_logs_and_traces/dotnet.md]:
```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Defaults to 0 if no trace. |
```

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 3]
```diff
@@ -75,61 +94,52 @@ For NLog version 4.6+:
...
+
+3. Configure mapped diagnostic context (MDC) to use the following injected keys:
+
+    | Required key   | Description                                  |
+    | -------------- | -------------------------------------------- |
+    | `dd.env`       | Globally configures the `env` for the tracer. Defaults to `""` if not set. |
+    | `dd.service`   | Globally configures the root service name. Defaults to the name of the application or IIS site name if not set.  |
+    | `dd.version`   | Globally configures `version` for the service. Defaults to `""` if not set.  |
+    | `dd.trace_id`  | Active trace ID during the log statement. Defaults to `0` if no trace.  |
+    | `dd.span_id`   | Active span ID during the log statement. Default to `0` if no trace. |
```
Comment (kayayarai) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:135]:
```suggestion
    | `dd.span_id`   | Active span ID during the log statement. Defaults to `0` if no trace. |
```

**Comment by kayayarai** [comment]
Yes, added

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 4]
```diff
@@ -75,61 +94,52 @@ For NLog version 4.6+:
...
-    <attribute name="exception" layout="${exception:format=ToString}" />
-
-    <attribute name="dd.env" layout="${mdc:item=dd.env}"/>
-    <attribute name="dd.service" layout="${mdc:item=dd.service}"/>
-    <attribute name="dd.version" layout="${mdc:item=dd.version}"/>
-    <attribute name="dd.trace_id" layout="${mdc:item=dd.trace_id}"/>
-    <attribute name="dd.span_id" layout="${mdc:item=dd.span_id}"/>
-  </layout>
 ```
+For additional examples, see the automatic trace ID injection projects [for NLog45][1] or [for NLog46][2] on GitHub.
```
Comment (zacharycmontoya) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:108]:
```suggestion
For additional examples, see the automatic trace ID injection projects using [NLog 4.5][1] or [NLog 4.6][2] on GitHub.
```
What do you think of this change?

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 5]
```diff
@@ -75,61 +94,52 @@ For NLog version 4.6+:
...
-    - `dd.env` Globally configured `env` for the tracer (defaults to `""` if not set)
-    - `dd.service` Globally configured root service name (defaults to the name of the application or IIS site name if not set)
-    - `dd.version` Globally configured `version` for the service (defaults to `""` if not set)
-    - `dd.trace_id` Active Trace ID during the log statement (defaults to `0` if no trace)
-    - `dd.span_id` Active Span ID during the log statement (defaults to `0` if no trace)
+1. Reference the [`Datadog.Trace` NuGet package][10] in your project.
+
+2. Use the `CorrelationIdentifier` API to retrieve correlation identifiers and add them to the log context while a span is active.
+
+3. Configure mapped diagnostic context (MDC) to use the following injected keys:
```
Comment (zacharycmontoya) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:127]: The setup with config files for log4net and NLog, and the `.Enrich.FromLogContext()` for Serilog are required for both automatic AND manual instrumentation. What do you think would be the best way to reference those from both sets of instructions?

File: content/en/tracing/connect_logs_and_traces/dotnet.md (hunk 4 above)
Comment (kayayarai) [reply content/en/tracing/connect_logs_and_traces/dotnet.md:108]: Yup, that looks much better.

File: content/en/tracing/connect_logs_and_traces/dotnet.md (hunk 5 above)
Comment (kayayarai) [reply content/en/tracing/connect_logs_and_traces/dotnet.md:127]: Got it, update incoming.

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 6]
```diff
@@ -75,61 +95,59 @@ For NLog version 4.6+:
...
-    <attribute name="dd.trace_id" layout="${mdc:item=dd.trace_id}"/>
-    <attribute name="dd.span_id" layout="${mdc:item=dd.span_id}"/>
-  </layout>
 ```
+For additional examples, see the automatic trace ID injection projects using [NLog 4.5][1] or [NLog 4.6][2] on GitHub.
 
-For NLog version lower than 4.0, there is no built-in JSON layout.
 
+[1]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog45Example/NLog.config
+[2]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog46Example/NLog.config
```
Comment (zacharycmontoya) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:113]:
```suggestion
[1]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog40Example/NLog.config
[2]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog45Example/NLog.config
[3]: https://github.com/DataDog/dd-trace-dotnet/blob/master/samples/AutomaticTraceIdInjection/NLog46Example/NLog.config
```
I just recently added an NLog 4.0 example 😄

File: content/en/tracing/connect_logs_and_traces/dotnet.md [hunk 7]
```diff
@@ -75,61 +95,59 @@ For NLog version 4.6+:
...
-    <attribute name="exception" layout="${exception:format=ToString}" />
-
-    <attribute name="dd.env" layout="${mdc:item=dd.env}"/>
-    <attribute name="dd.service" layout="${mdc:item=dd.service}"/>
-    <attribute name="dd.version" layout="${mdc:item=dd.version}"/>
-    <attribute name="dd.trace_id" layout="${mdc:item=dd.trace_id}"/>
-    <attribute name="dd.span_id" layout="${mdc:item=dd.span_id}"/>
-  </layout>
 ```
+For additional examples, see the automatic trace ID injection projects using [NLog 4.5][1] or [NLog 4.6][2] on GitHub.
```
Comment (zacharycmontoya) [inline content/en/tracing/connect_logs_and_traces/dotnet.md:109]:
```suggestion
For additional examples, see the automatic trace ID injection projects using [NLog 4.0][1], [NLog 4.5][2], or [NLog 4.6][3] on GitHub.
```

I just recently added an NLog 4.0 example 😄

**Review by zacharycmontoya — APPROVED** [review]
Thanks for driving these changes @kayayarai ! I think the result is much clearer for our customers

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**